nornir_nautobot/plugins/tasks/dispatcher/mikrotik_routeros.py

```diff
@@ -0,0 +1,98 @@
+from nornir.core.task import Result, Task
+from nornir_netmiko.tasks import netmiko_send_command
+
+from nornir_nautobot.exceptions import NornirNautobotException
+from nornir_nautobot.utils.helpers import make_folder
+from .default import NetmikoNautobotNornirDriver as DefaultNautobotNornirDriver
+
+GET_VERSION_COMMAND = "system resource print"
+GET_MAJOR_VERSION_REGEX = re.compile(r"version:\s+(\d+)\.\d+\.\d+")
+GET_CONFIG_ROS = "export terse"
+NETMIKO_DEVICE_TYPE = "mikrotik_routeros"
+
+
+class NautobotNornirDriver(DefaultNautobotNornirDriver):
+    """Driver for Mikrotik Router OS."""
+
+    @staticmethod
+    def get_config(task: Task, logger, obj, backup_file: str, remove_lines: list, substitute_lines: list) -> Result:
+        """Get the latest configuration from the device using Netmiko. Overrides default get_config.
+        
+        This accounts for Mikrotik Router OS config scrubbing behavior since ROS >= 7.X.
+
+        Args:
+            task (Task): Nornir Task.
+            logger (NornirLogger): Custom NornirLogger object to reflect job results (via Nautobot Jobs) and Python logger.
+            obj (Device): A Nautobot Device Django ORM object instance.
+            remove_lines (list): A list of regex lines to remove configurations.
+            substitute_lines (list): A list of dictionaries with to remove and replace lines.
+
+        Returns:
+            Result: Nornir Result object with a dict as a result containing the running configuration
+                { "config: <running configuration> }
+        """
+        task.host.platform = NETMIKO_DEVICE_TYPE  # Patch for platform_slug mapping (temporal)
+        logger.log_debug(f"Analyzing Software Version for {task.host.name} on {task.host.platform}")
+        command = GET_VERSION_COMMAND
+        try:
+            result = task.run(task=netmiko_send_command, command_string=GET_VERSION_COMMAND)
+        except NornirSubTaskError as exc:
+            if isinstance(exc.result.exception, NetmikoAuthenticationException):
+                logger.log_failure(obj, f"Failed with an authentication issue: `{exc.result.exception}`")
+                raise NornirNautobotException(f"Failed with an authentication issue: `{exc.result.exception}`")
+
+            if isinstance(exc.result.exception, NetmikoTimeoutException):
+                logger.log_failure(obj, f"Failed with a timeout issue. `{exc.result.exception}`")
+                raise NornirNautobotException(f"Failed with a timeout issue. `{exc.result.exception}`")
+
+            logger.log_failure(obj, f"Failed with an unknown issue. `{exc.result.exception}`")
+            raise NornirNautobotException(f"Failed with an unknown issue. `{exc.result.exception}`")
+
+        if result[0].failed:
+            return result
+
+        search_result = re.search(GET_MAJOR_VERSION_REGEX, result[0].result)
+        major_version = search_result.group(1)
+
+        command = GET_CONFIG_ROS
+        if major_version > "6":
```

Comment:
I am not sure how we should handle this. I don't have an answer, but just want to raise the question .

Reply:
I'm good with the idea for the concept of adding commands based on versions, if that is matches what is needed.

We could probably do this with a native Python module though rather than regex?

Reply:
I can probably get consistent results across versions by splitting the output multiple times. Will test and update if that's ok.